pytorch_tabnet/tab_network.py

```diff
@@ -117,10 +122,13 @@ def __init__(self, input_dim, output_dim,
                                           n_glu_independent=self.n_independent,
                                           virtual_batch_size=self.virtual_batch_size,
                                           momentum=momentum)
-            attention = AttentiveTransformer(n_a, self.input_dim,
-                                             virtual_batch_size=self.virtual_batch_size,
-                                             momentum=momentum,
-                                             mask_type=self.mask_type)
+            attention = AttentiveTransformer(
+                n_a,
+                len(self.feature_embed_widths) if self.feature_embed_widths else self.input_dim,
```

Comment:
If I understand correctly don't you always have `len(self.feature_embed_widths)==self.input_dim` ?

Reply:
No - since `TabNetNoEmbeddings` is just the portion of the network after embeddings have been done, the `input_dim` is the post-embedding dimension.